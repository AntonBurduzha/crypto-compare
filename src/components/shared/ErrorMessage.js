// @flow
import React from 'react';
import { Row, Col } from 'antd';
import type { Element } from 'react';

export const ErrorMessage = (): Element<any> => {
  return (
    <Row type="flex" justify="center" align="middle">
      <Col span={12}>
        <h3 style={{ textAlign: 'center' }}>Oups! Error occurs, please try again later.</h3>
      </Col>
    </Row>
  );
}
