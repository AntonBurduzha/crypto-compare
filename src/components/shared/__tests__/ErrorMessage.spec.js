import React from 'react';
import { shallow } from 'enzyme';
import { ErrorMessage } from '../ErrorMessage';

describe('ErrorMessage component', () => {
  const wrap = shallow(<ErrorMessage/>);

  it('renders without crashing', () => {
    expect(wrap).toBeDefined();
  });

  it('has default text', () => {
    expect(wrap.find('h3').text()).toEqual('Oups! Error occurs, please try again later.');
  });

  it('matches snapshot', () => {
    expect(wrap).toMatchSnapshot();
  })
});
