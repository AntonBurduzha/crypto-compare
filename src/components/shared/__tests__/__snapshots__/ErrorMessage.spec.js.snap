// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorMessage component matches snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "align": "middle",
      "children": <Col
        span={12}
>
        <h3
                style={
                        Object {
                                "textAlign": "center",
                              }
                }
        >
                Oups! Error occurs, please try again later.
        </h3>
</Col>,
      "gutter": 0,
      "justify": "center",
      "type": "flex",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <h3
          style={
                    Object {
                              "textAlign": "center",
                            }
          }
>
          Oups! Error occurs, please try again later.
</h3>,
        "span": 12,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Oups! Error occurs, please try again later.",
          "style": Object {
            "textAlign": "center",
          },
        },
        "ref": null,
        "rendered": "Oups! Error occurs, please try again later.",
        "type": "h3",
      },
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "align": "middle",
        "children": <Col
          span={12}
>
          <h3
                    style={
                              Object {
                                        "textAlign": "center",
                                      }
                    }
          >
                    Oups! Error occurs, please try again later.
          </h3>
</Col>,
        "gutter": 0,
        "justify": "center",
        "type": "flex",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <h3
            style={
                        Object {
                                    "textAlign": "center",
                                  }
            }
>
            Oups! Error occurs, please try again later.
</h3>,
          "span": 12,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Oups! Error occurs, please try again later.",
            "style": Object {
              "textAlign": "center",
            },
          },
          "ref": null,
          "rendered": "Oups! Error occurs, please try again later.",
          "type": "h3",
        },
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
